test/operators.jl

```diff
@@ -0,0 +1,740 @@
+        for sd in dual_meshes_1D
+            @test all(dec_boundary(i, sd) .== ∂(i, sd))
+        end
+    end
+
+    for i in 1:2
+        for sd in dual_meshes_2D
+            @test all(dec_boundary(i, sd) .== ∂(i, sd))
+        end
+    end
+end
+
+@testset "Dual Derivative" begin
+    for i in 0:0 
+        for sd in dual_meshes_1D
+            @test all(dec_dual_derivative(i, sd) .== dual_derivative(i, sd))
+        end
+    end
+
+    for i in 0:1
+        for sd in dual_meshes_2D
+            @test all(dec_dual_derivative(i, sd) .== dual_derivative(i, sd))
+        end
+    end
+end
+
+#TODO: For hodge star 1, the values seems to be extremely close yet not quite equal
+@testset "Diagonal Hodge" begin
+    for i in 0:1
+        for sd in dual_meshes_1D
+            @test all(isapprox.(dec_hodge_star(Val{i}, sd, DiagonalHodge()), hodge_star(i, sd, DiagonalHodge()); rtol = 1e-15))
+        end
+    end
+
+    for i in 0:2
+        for sd in dual_meshes_2D
+            @test all(isapprox.(dec_hodge_star(Val{i}, sd, DiagonalHodge()), hodge_star(i, sd, DiagonalHodge()); rtol = 1e-15))
+        end
+    end
+end
+
+#TODO: For inv hodge star 1, the values seems to be extremely close yet not quite equal
+@testset "Inverse Diagonal Hodge" begin
+    for i in 0:1
+        for sd in dual_meshes_1D
+            @test all(isapprox.(dec_inv_hodge(Val{i}, sd, DiagonalHodge()), inv_hodge_star(i, sd, DiagonalHodge()); rtol = 1e-15))
+        end
+    end
+
+    for i in 0:2
+        for sd in dual_meshes_2D
+            @test all(isapprox.(dec_inv_hodge(Val{i}, sd, DiagonalHodge()), inv_hodge_star(i, sd, DiagonalHodge()); rtol = 1e-15))
+        end
+    end
+end
+
+@testset "Geometric Hodge" begin
+    for i in 0:1
+        for sd in dual_meshes_1D
+            @test all(isapprox.(dec_hodge_star(Val{i}, sd, GeometricHodge()), hodge_star(i, sd, GeometricHodge()); rtol = 1e-15))
+        end
+    end
+
+    for i in [0, 2]
+        for sd in dual_meshes_2D
+            @test all(isapprox.(dec_hodge_star(Val{i}, sd, GeometricHodge()), hodge_star(i, sd, GeometricHodge()); rtol = 1e-15))
+        end
+    end
+
+    # TODO: Why does this test require atol, not rtol, to reasonably pass?
+    for i in [1]
+        for sd in dual_meshes_2D
+            @test all(isapprox.(dec_hodge_star(Val{i}, sd, GeometricHodge()), hodge_star(i, sd, GeometricHodge()); atol = 1e-15))
+        end
+    end
+
+end
+
+@testset "Inverse Geometric Hodge" begin
+    for i in 0:1
+        for sd in dual_meshes_1D
+            @test all(isapprox.(dec_inv_hodge(Val{i}, sd, GeometricHodge()), inv_hodge_star(i, sd, GeometricHodge()); rtol = 1e-15))
+        end
+    end
+
+    for i in [0, 2]
+        for sd in dual_meshes_2D
+            @test all(isapprox.(dec_inv_hodge(Val{i}, sd, GeometricHodge()), inv_hodge_star(i, sd, GeometricHodge()); rtol = 1e-15))
+        end
+    end
+
+    for i in 1:1
+        for sd in dual_meshes_2D[1:end-1]
+            V_1 = rand(ne(sd))
```

Comment:
Add the `@show` macro in front of `V_1` and check the Github actions output between different commits.